third-party/realdds/src/dds-option.cpp

```diff
@@ -32,14 +34,22 @@ dds_option::dds_option( nlohmann::json const & j, const std::string & owner_name
 
     if( index != j.size() )
         DDS_THROW( runtime_error, "expected end of json at index " + std::to_string( index ) );
+}
+
 
-    _owner_name = owner_name;
+void dds_option::init_stream( std::shared_ptr< dds_stream_base > const & stream )
```

Comment:
The name `init_stream` sounds not right, you are not performing initialization on the stream. Maybe `set_stream` or `associate_with_stream`?

Reply:
This mirrors the way we arranged it in `dds_stream_profile`. As I recall we wanted to convey that you can do this only once so it was part of initialization.